File: backend-server/app/Models/CrawledPages.php
<?php

namespace App\Models;

use DateTimeInterface;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Ramsey\Uuid\Uuid;
use Ramsey\Uuid\UuidInterface;

class CrawledPages extends Model
{
    use HasFactory;

    protected $table = 'crawled_pages';

    protected $fillable = [
        'id',
        'chatbot_id',
        'website_data_source_id',
        'url',
        'title',
        'status_code',
    ];

    public function getId(): UuidInterface
    {
        return Uuid::fromString($this->id);
    }

    public function getChatbotId(): UuidInterface
    {
        return Uuid::fromString($this->chatbot_id);
    }

    public function getWebsiteDataSourceId(): UuidInterface
    {
        return Uuid::fromString($this->website_data_source_id);
    }

    public function getUrl(): string
    {
        return $this->url;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }


    public function getStatusCode(): string
    {
        return $this->status_code;
    }

    public function setId(UuidInterface $id): void
    {
        $this->id = $id;
    }

    public function setChatbotId(UuidInterface $chatbotId): void
    {
        $this->chatbot_id = $chatbotId;
    }

    public function setWebsiteDataSourceId(UuidInterface $websiteDataSourceId): void
    {
        $this->website_data_source_id = $websiteDataSourceId;
    }

    public function setUrl(string $url): void
    {
        $this->url = $url;
    }

    public function setTitle(?string $title): void
    {
        $this->title = $title;
    }

    public function setStatusCode(?string $statusCode): void
    {
        $this->status_code = $statusCode;
    }

    public function getCreatedAt(): DateTimeInterface
    {
        return $this->created_at;
    }
}

File: dj_backend_server/web/models/crawled_pages.py
from django.db import models
import uuid
from .website_data_sources import WebsiteDataSource
from web.models.chatbot import Chatbot

class CrawledPages(models.Model):
    id = models.AutoField(primary_key=True)
    chatbot_id = models.CharField(max_length=36, null=True)
    website_data_source = models.ForeignKey(WebsiteDataSource, on_delete=models.CASCADE, related_name='crawled_pages')
    url = models.CharField(max_length=255)
    title = models.CharField(max_length=255, null=True)
    status_code = models.CharField(max_length=255, null=True)
    aws_url = models.TextField(null=True)
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)
    content_file= models.CharField(max_length=255, null=True)

    def get_id(self):
        return self.id

    def get_chatbot_id(self):
        return self.chatbot_id

    def get_website_data_source_id(self):
        return self.website_data_source_id

    def get_url(self):
        return self.url

    def get_title(self):
        return self.title

    def get_status_code(self):
        return self.status_code

    def set_id(self, _id):
        self.id = _id

    def set_chatbot_id(self, chatbot_id):
        self.chatbot_id = chatbot_id

    def set_website_data_source_id(self, website_data_source_id):
        self.website_data_source_id = website_data_source_id

    def set_url(self, url):
        self.url = url

    def set_title(self, title):
        self.title = title

    def set_status_code(self, status_code):
        self.status_code = status_code

    def get_created_at(self):
        return self.created_at

    class Meta:
        db_table = 'crawled_pages'  # Replace 'crawled_pages' with the actual table name in the database
